### ISSUE TITLE
[Bug] Backup fails if the folder to be compressed is not deleted (from the previous backup) + An error is thrown which completely blocks the backup function

### ISSUE THREAD

**Issue by zFerry98** [hmailserver/hmailserver#445]
I had changed some configurations on the internet IP range (I required smtp auth for those who send emails to me🤷‍♂️), which is why I found myself sifting through all the software to find them (folders included) and I discover that I had left the debug active and the beauty of 144gb of Logs! (that are not included in the backup but despite this, they are counted in the warning inside the backup log: You have exceeded 1.5gb)
I was at the point of blaming PostgreeSQl and was about to switch to MySQL5 (because 8 is among the versions that remove 32bit Plugin support).
So I start a backup ... well, the result was that I found out when it "migrated" to MySQL that there was no backup of that day.
So I went back to Postgree and went through the procedure again.

Here are the error logs:

<details>
 <summary>The 3 logs:</summary>

```
"ERROR"	8604	"2022-09-15 10:21:25.696"	"Severity: 1 (Critical), Code: HM5014, Source: BackupManager::OnBackupFailed, Description: BACKUP ERROR: Backup or restore operation is already started"
"ERROR"	11420	"2022-09-15 10:23:25.795"	"Severity: 2 (High), Code: HM4208, Source: ExceptionHandler::Run, Description: An error occured while executing 'Task-BackupTask', Error code: 80, Message: boost::filesystem::copy_file: File esistente: "C:\Program Files (x86)\hMailServer\Data\xxxxxxxxxxxxxx\admin\00\{008EFD9E-8357-4493-ADC7-F44389190137}.eml", "B:\\SERVER\\Backup\\hMailServer\\Full\DataBackup\xxxxxxxxxxxxxx\admin\00\{008EFD9E-8357-4493-ADC7-F44389190137}.eml""
"ERROR"	11420	"2022-09-15 10:23:26.107"	"Severity: 1 (Critical), Code: HM5519, Source: StackLogger::Log, Description: An error has been detected. A mini dump has been written to C:\Program Files (x86)\hMailServer\Logs\minidump_2022-09-15 102325_{35E693FC-FF46-406A-8123-7A8D2AE84488}.dmp"
```

```
2022-09-15 10:21:25.694	Backup started
﻿2022-09-15 10:21:25.695	BACKUP ERROR: Backup or restore operation is already started
﻿2022-09-15 10:23:25.208	Backup started
﻿2022-09-15 10:23:25.209	Loading backup settings....
﻿2022-09-15 10:23:25.664	Backing up domains...
﻿2022-09-15 10:23:25.668	Backing up account 
﻿2022-09-15 10:23:25.682	Backing up account 
﻿2022-09-15 10:23:25.689	Backing up account 
﻿2022-09-15 10:23:25.706	Backing up account  
﻿2022-09-15 10:23:25.708	Backing up account  
﻿2022-09-15 10:23:25.711	Backing up account  
﻿2022-09-15 10:23:25.716	Backing up account  
﻿2022-09-15 10:23:25.723	Backing up account 
﻿2022-09-15 10:23:25.730	Backing up account
﻿2022-09-15 10:23:25.739	Backing up account
﻿2022-09-15 10:23:25.743	Backing up account
﻿2022-09-15 10:23:25.790	Backing up data directory...
﻿﻿2022-09-15 10:25:20.975	Backup started
﻿2022-09-15 10:25:20.976	Loading backup settings....
﻿2022-09-15 10:25:21.380	Backing up domains...
﻿2022-09-15 10:25:21.384	Backing up account
﻿2022-09-15 10:25:21.395	Backing up account
﻿2022-09-15 10:25:21.403	Backing up account
﻿2022-09-15 10:25:21.418	Backing up account
﻿2022-09-15 10:25:21.420	Backing up account
﻿2022-09-15 10:25:21.422	Backing up account
﻿2022-09-15 10:25:21.426	Backing up account
﻿2022-09-15 10:25:21.433	Backing up account
﻿2022-09-15 10:25:21.438	Backing up account
﻿2022-09-15 10:25:21.447	Backing up account
﻿2022-09-15 10:25:21.451	Backing up account
﻿2022-09-15 10:25:21.495	Backing up data directory...
﻿2022-09-15 10:25:35.035	Backing up settings...
﻿2022-09-15 10:25:35.069	Writing XML file...
﻿2022-09-15 10:25:35.303	Compressing message files...
﻿2022-09-15 10:25:38.468	Backup completed successfully
```

```
"ERROR"	8604	"2022-09-15 10:21:25.696"	"Severity: 1 (Critical), Code: HM5014, Source: BackupManager::OnBackupFailed, Description: BACKUP ERROR: Backup or restore operation is already started"
"APPLICATION"	17292	"2022-09-15 10:22:40.522"	"Stopping servers..."
"APPLICATION"	17292	"2022-09-15 10:22:41.038"	"Servers stopped."
"APPLICATION"	16508	"2022-09-15 10:22:43.072"	"Starting servers..."
"APPLICATION"	16508	"2022-09-15 10:22:43.082"	"Servers started."
"ERROR"	11420	"2022-09-15 10:23:25.795"	"Severity: 2 (High), Code: HM4208, Source: ExceptionHandler::Run, Description: An error occured while executing 'Task-BackupTask', Error code: 80, Message: boost::filesystem::copy_file: File esistente: "C:\Program Files (x86)\hMailServer\Data\xxxxxxxxxxxxxx\admin\00\{008EFD9E-8357-4493-ADC7-F44389190137}.eml", "B:\\SERVER\\Backup\\hMailServer\\Full\DataBackup\xxxxxxxxxxxxxx\admin\00\{008EFD9E-8357-4493-ADC7-F44389190137}.eml""
"ERROR"	11420	"2022-09-15 10:23:26.107"	"Severity: 1 (Critical), Code: HM5519, Source: StackLogger::Log, Description: An error has been detected. A mini dump has been written to C:\Program Files (x86)\hMailServer\Logs\minidump_2022-09-15 102325_{35E693FC-FF46-406A-8123-7A8D2AE84488}.dmp"
"APPLICATION"	21852	"2022-09-15 10:24:47.598"	"Stopping servers..."
"APPLICATION"	21852	"2022-09-15 10:24:47.616"	"Servers stopped."
"APPLICATION"	14760	"2022-09-15 10:24:50.142"	"Starting servers..."
"APPLICATION"	14760	"2022-09-15 10:24:50.154"	"Servers started."
```
</details>

Then, the backup starts, arrives at the "DATA" folder and goes into error, remaining in the "backup in progress" state.
This state prevents you from starting any more backups.
The only way to restart them is to restart the service.
Obviously it does not solve the problem (which is the presence of the contents of the "DATA" directory, which had to be deleted after the end of the previous backup).
Deleting by hand and restarting the service, as from the 2nd log, it returns to work properly.

### The bug:

- If the software crashes while making a backup, you can no longer perform backups until you clean the folder where those backups take place (which is SERIOUS if you have a schedule and don't notice it, you end up with months of backups never made);

_Corrected by the answer:
https://github.com/hmailserver/hmailserver/issues/445#issuecomment-1247879554_
-----------------------------
Furthermore, the logs have a lot of sensitive information, it would be optimal to find a way to filter them in a log to be able to share on github in the future.

**Comment by mstenz** [comment]
To me backups also constantly failing. (at leat once a month). There are different reasons for this. For example when emails get deleted while they are backed up. This happens fore sure more frequently in a high volume/users environment.

**Comment by zFerry98** [comment]
> To me backups also constantly failing. (at leat once a month). There are different reasons for this. For example when emails get deleted while they are backed up. This happens fore sure more frequently in a high volume/users environment.

In my case we are a few people and all in the same country, so I start the planning at 4 am (for 30mb it takes 10 seconds).
I was opting for a more elaborate and "manual" solution.
The problem is that damn Database ... whether it is MySQL or PostGreeSQL, they are not locked and backed up by external tools if you do not first stop the software that accesses them (hMailS).

**Comment by RvdHout** [comment]
Mmm, not sure everything/anything what is said here is entirely true

1. logs are **not** included in the backup
2. backups take a long time to complete, OK it lacks a progress indicator or something but most likely you interfered the backup progress before it completed

**Comment by zFerry98** [comment]
> Mmm, not sure everything/anything what is said here is entirely true
> 
> 1. logs are **not** included in the (build-in) backup
> 2. (build-in) backups take a long time to complete, OK it lacks a progress indicator or something but most likely you interfered the backup progress before it completed
> 
> My backups are approx 15GB big, and take around a hour to complete, eg: backup of database, backup of settings, backup of data and finally the creation of the 7z archive

Maybe they are not included in 7z but in the log it tells you that you have exceeded 1.5gb recommended (the only folder that exceeds 1gb were the 144gb logs). By removing the only folder that exceeded 1gb (logs), the warning in the log that I had over 1.5gb of Data disappeared (as you can see from the logs I posted to create the issue).
Anyway thanks, I correct above

**Comment by RvdHout** [comment]
What version are you using?
I think that is just a warning, not? 
I never seen the exceeded 1.5gb recommended warning....but i am using a [custom build](https://www.hmailserver.com/forum/viewtopic.php?p=241830#p241830) that doesn't have this limitation

> Fix: Removed warning if backup was more than 1,5GB and 15GB limit. There's no longer a recommended max-size - the time will vary with the installation size. [issue #69](https://github.com/hmailserver/hmailserver/issues/69)

**Comment by zFerry98** [comment]
> What version of hMailServer are you using? I think that is just a warning, not?
> 
> I never seen the exceeded 1.5gb recommended warning....but i am using a [custom build](https://www.hmailserver.com/forum/viewtopic.php?p=241830#p241830) that doesn't have this limitation
> 
> > .7 Fix: Removed warning if backup was more than 1,5GB and 15GB limit. There's no longer a recommended max-size - the time will vary with the installation size. [issue #69](https://github.com/hmailserver/hmailserver/issues/69)
> 
> I checked one of my backups, there are no log files included

5.6.9-B2602 
![image](https://user-images.githubusercontent.com/17975351/190380682-5ed2cc7e-afcb-4288-8ce1-77ee7bcf54c8.png)

Now I go to un7zip these 144gb compressed into 9gb and see if I can pull out the exact line (hoping it wasn't in the logs after the archive, because those have been removed)

**Comment by RvdHout** [comment]
hmailserver_backup.log using the custom build i referred to earlier (backup took little less then an hour, backup size=~14GB ) 

```
2022-07-31 22:39:17.009	Backup started
﻿2022-07-31 22:39:17.024	Loading backup settings....
﻿2022-07-31 22:39:17.040	Backing up domains...
﻿2022-07-31 22:39:17.056	Backing up account info@...
* Repeated may times for every account 
﻿2022-08-31 22:49:33.134	Backing up data directory...
﻿2022-08-31 23:26:51.434	Backing up settings...
﻿2022-08-31 23:26:51.466	Writing XML file...
﻿2022-08-31 23:26:54.116	Compressing message files...
﻿2022-09-01 00:14:25.465	Backup completed successfully

```

**Comment by zFerry98** [comment]
> What version of hMailServer are you using? I think that is just a warning, not?
> 
> I never seen the exceeded 1.5gb recommended warning....but i am using a [custom build](https://www.hmailserver.com/forum/viewtopic.php?p=241830#p241830) that doesn't have this limitation
> 
> > .7 Fix: Removed warning if backup was more than 1,5GB and 15GB limit. There's no longer a recommended max-size - the time will vary with the installation size. [issue #69](https://github.com/hmailserver/hmailserver/issues/69)
> 
> I checked one of my backups, there are no log files included

![image](https://user-images.githubusercontent.com/17975351/190382421-5d6e542f-7c26-47f9-a178-12ae51fa76a9.png)

**Comment by RvdHout** [comment]
It says /Data directory, last time i checked the log files are not in the /Data directory but in /Logs

**Comment by zFerry98** [comment]
> It says /Data directory, last time i checked the log files are not in the /Data directory but in /Logs, so doubt the warning is caused by the log files in /Logs directory

![image](https://user-images.githubusercontent.com/17975351/190383091-625cda0b-8698-4d43-b45f-9030b09fc80f.png)
I doubt it reaches 100mbs ...
And as I said above, after removing all the logs (that screen is just I had finished 7z after 7 hours). The Warn from the logs is gone.

**Comment by zFerry98** [comment]
I say it again:
The backup starts a function that controls the size of the so-called "DATA" directory ... In its magical function, it counts the 144gb Logs folder BUT (as you said, I also confirm now) IT IS NOT added to the final backup (which includes the Data folder and the .xml of the server).
Where does Warn come from if the Data folder is 30mb? and all other folders EXCEPT "Logs" are also under 100mb?

![image](https://user-images.githubusercontent.com/17975351/190384586-6a86f72a-abdd-49a7-8038-d5d9f259cb72.png)

**Comment by RvdHout** [comment]
https://github.com/hmailserver/hmailserver/commit/93663addca13ae77b464112538bfdc56e8ffab87

As you can see in above changes where to limitations where removed, It took the size of all messages, sorry to disappoint you, but it really ain't the logs

**Comment by zFerry98** [comment]
> I say it again: The backup starts a function that controls the size of the so-called "DATA" directory ... In its magical function, it counts the 144gb Logs folder BUT (as you said, I also confirm now) IT IS NOT added to the final backup (which includes the Data folder and the .xml of the server). Where does Warn come from if the Data folder is 30mb? and all other folders EXCEPT "Logs" are also under 100mb?
> 

> [93663ad](https://github.com/hmailserver/hmailserver/commit/93663addca13ae77b464112538bfdc56e8ffab87)
> 
> As you can see in above changes where this limitation was removed, It took the size of all messages, sorry to disappoint you, but it really ain't the logs

Now we understand each other!
And how do you explain to me that after deleting the Logs folder the warn is gone? (since the data folder and emails I have not touched and they are all still there).

I'm not trying to offend but you DEV's assume that everyone knows how to program and read the code of everything in the world ...

I use the software as an external and report the issue with as many details as possible FROM MY POINT OF VIEW, I repeat:
## **AFTER REMOVING THE 144GB "LOGS" FOLDER, THE WARN IS DISAPPEARED FROM THE LOG!**

**Comment by RvdHout** [comment]
You can become mad, but i just proven to you the log files are not included in this count!

**Comment by zFerry98** [comment]
> You can become mad, but i just proven to you the log files are not included in this count!



> You can become mad all you want, but i just proven to you the log files are not included in this count!

I'm mad Yes!

Why do you keep blaming me for saying falsehood!
And I repeat to you:
I AM NOT A PROGRAMMER (because I would not even open the issue, I would make a pull request with the fix), I EXPLAINED WHAT I HAVE DONE and the answer is right and I tell you: "ok, thanks", but don't give me a fake!
Because of internal software problems in the code THAT ARE NOT MENTIONED IN THE DOCUMENTATION ... WHAT A FUCK DO I KNOW ?!

Come on, let's finish it here.
The bug remains, if the backup crashes it will not restart until you restart the service and you have to read the logs because in the GUI it is not said that the backup has gone wrong (and that you also have to restart the service to make it work again).
This is because it is not checked that in the Backup folder, the previous backup has finished correctly with relative cleaning of the "temporary" folder that it uses to make the .7z.

**Comment by RvdHout** [comment]
The backup did not crash, you interrupted it because you saw a WARNING!!! **(Warning != Error)**

I just stating the facts here, the log files directory is not included in the backup, the total size of the backup is calculated by the sum of the mailboxsize column of the hm_messages database table

**Comment by zFerry98** [comment]
The answer in a serious context was:
Ah! sorry, it seems that in OUR software message lengths are counted as well as file size (or whatever).
The question related to the logs is unexpected, can you give us more details on what you did and how?

![image](https://user-images.githubusercontent.com/17975351/190389551-93caa386-a637-4126-9b2d-f81dab8fbf31.png)
![image](https://user-images.githubusercontent.com/17975351/190389734-47e3b827-907c-493d-a419-13c37d05f167.png)
It is certainly not the size of the files containing the messages (if they are in the given directory).
![image](https://user-images.githubusercontent.com/17975351/190390693-f88ce62c-34a9-4c6e-97ab-30609e1b0e88.png)
Probably not even in the DB

-------------------------------------------
> The backup did not crash, you interrupted it because you saw a WARNING!!! **(Warning != Error)**
> 
> I just stating the facts here, the log files directory is not included in the backup, the total size of the backup is calculated by the sum of the mailboxsize column of the hm_messages database table
> 
> These are my answers/findings, if you don't like them, please do not reply any further

Then you did it on purpose!
THAT FUCKING ERROR IS REGISTERED BY THE SOFTWARE AND IS DUE TO AN EXTERNAL INTERRUPTION !!
NOT MADE BY ME BECAUSE I SEE DRAGONS RANDOMLY AND SO I READ THE LOGS AND THEN I SAY: "LET'S GO TO DO A '_CLOSE PROCCESS_' BECAUSE I READ IN THE LOG" WARN "OF MY COCK !!!".

Are we joking?!
I'm wasting time for you who don't understand shit.

Anyway, this software is abandoned... In the end I solved the problem and I wasted myself in giving a temporary solution to those who are less good in this field.

**Comment by RvdHout** [comment]
Well spoken, I am really, really impressed....NOT!

**Comment by zFerry98** [comment]
> Well spoken, I am really, really impressed....NOT!

When you are no longer able to challenge the truth in the face, always change the subject!
I keep giving facts and saying that I did NOT close the software during the backup!
You keep talking bullshit and agreeing with yourself!

Maybe I have to repeat for the 6th time with the insults, so you understand ... (then I get punished, because reading and understanding who is right is out of the reach of this shitty world):
IF THE FUCK OF SOFTWARE DOES NOT FINISH THE BACKUP (TYPE THE CURRENT GOES AWAY WHILE), THE BACKUP FUNCTION NO LONGER GOES BECAUSE THERE ARE THE RESIDUES OF THE PREVIOUS BACKUP!
ALSO, YOU DON'T UNDERSTAND WHAT IT CALCULATES FOR THE BACKUP SIZE!
GIVEN THAT I HAVE POSTED THE SCREEN AND I DO NOT SEE ANYTHING THAT EXCEEDS 1GB ADDING ALL THE VALUES (EXCLUDING THE FANTOMATIC LOGS FOLDER WHICH SEEMS TO HAVE SOMETHING RELATIVE AND MAYBE NOT WRITTEN IN THAT .CPP OF SHIT THAT IS NOT THE ONLY ONE!)

**Comment by RvdHout** [comment]
Keep crying... simply loving it 😂

If you think you can use such tone to someone who only is trying to show (and prove) you the problem wasn't the log files you are wrong big time, and you dare to say something on the words i chosen to try explain bits and pieces....

get lost!

**Comment by zFerry98** [comment]
> Keep crying... simply loving it 😂
> 
> If you think you can use such tone to someone who only is trying to show (and prove) you the problem wasn't the log files you are wrong big time, and you dare to say something on the words i chosen to try explain bits and pieces....
> 
> get lost!

I think you have not understood anything as usual ...
If you just minimally thought I was "crying".

I solved the problem before opening the issue: because unlike you I am what YOU call crying because the software you write doesn't work and you blame others instead of re-reading your crap code.
The simplest example are the morons who don't know anything about SSL / TLS and blame that the certificates don't work because of us and don't realize that they are the ones who have written the code wrong (you do the hand after a while).


"to show (and prove) you the problem wasn't the log files" BIT ON BIT WRITTEN demonstration that you just don't understand shit!

7th time:
https://github.com/hmailserver/hmailserver/issues/445#issuecomment-1247879554

> Mmm, not sure everything/anything what is said here is entirely true

I go over it, maybe you are not English and the translator wrote badly.
  Maybe you meant that there are some wrong processes in my reasoning because in the code it doesn't say that the logs folder is counted (but you're a jerk and you haven't even apologized for it ...)
 But hey!
I am GOD and I am automatically able to read anything and I have read all the code of this software, so I obviously knew about this detail!
> 
> 1. logs are **not** included in the (build-in) backup

And here I agreed (damn me ...) and re-corrected the first post (always without you having yet to explain why the warn disappeared from the logs only after removing the only files that exceeded 1gb )
>
> 2. (build-in) backups take a long time to complete, OK it lacks a progress indicator or something but most likely **you interfered the backup progress before it completed**

And here I get pissed off because you are already blaming others.
You didn't have the slightest decency to say "ah, maybe hMailServer really crashed or the power went out just as the backup was going"
> 
> My backups are approx 15GB big, and take around a hour to complete, eg: backup of domains, backup of settings, backup of messages and finally the creation of the 7z archive

--------------------------------
https://github.com/hmailserver/hmailserver/issues/445#issuecomment-1247887582



> > Mmm, not sure everything/anything what is said here is entirely true
> > 
> > 1. logs are **not** included in the (build-in) backup
> 
> Maybe they are not included in 7z but in the log it tells you that you have exceeded 1.5gb recommended (the only folder that exceeds 1gb were the 144gb logs). By removing the only folder that exceeded 1gb (logs), the warning in the log that I had over 1.5gb of Data disappeared (as you can see from the logs I posted to create the issue). Anyway thanks, I correct above.

Here I answered in the best and decent way to a person who has already made a mistake from the beginning and accused me (big mistake, I had to insult you right away ...).
AS YOU HAVE NO IDEA of what happened, you stuck to this fucking .cpp to back up your bullshit.
Here I have literally placed every evidence that the numbers to be counted, added together (DB + Data folder) do not reach 1gb.
You still haven't shown me where the warn came from (since it disappeared since I removed the 144gb logs) but according to your reasoning it should still be present since the logs folder belongs, I have not touched anything (what that the 2nd log is not seen in the first post where there is no more warn ...)

> 
> > 2. (build-in) backups take a long time to complete, OK it lacks a progress indicator or something but most likely you interfered the backup progress before it completed
> > 
> > My backups are approx 15GB big, and take around a hour to complete, eg: backup of database, backup of settings, backup of data and finally the creation of the 7z archive
> 
> "you interfered the backup progress before it completed"
> 
> Well, I'm not Windows itself and neither is the electricity or the RedScreen of the Socket2 of the server's mobo...

Here I tried to stop myself from fucking you ... if I didn't hold back maybe you immediately understood your mistake


If it doesn't get into your head like this ... you're a lost case.
Now I have nothing more to complain about (and I saved everything, you can't hide editing shit)

**Comment by RvdHout** [comment]
🖕

**Comment by zFerry98** [comment]
> 🖕

And here we return to (https://github.com/hmailserver/hmailserver/issues/445#issuecomment-1247999293) where those who give the truth in the face can not help but change the subject.

**Comment by RvdHout** [comment]
You must be a retard, so sorry everything i tried to explain clearly made no sense, i apologize for that...
The only one here not accepting the truth and facts is you, i only tried to point that out to you and all i got back are insults and explicit language, poor you.

I don't have to give you all the answers, hence i am not even the developer of this software and am just like you a end-user

**Comment by RvdHout** [comment]
> Did you go to see what the function does?
> 
> `SQLCommand command("select sum(messagesize) / 1024 / 1024 as mailboxsize from hm_messages");`

I already referred to that code didn't I? The fact you didn't bother to look at anything i wrote doesn't make you right....clown!

**Comment by zFerry98** [comment]
now without a fucking excuse, I'm going to solve another big problem... sending the SIGINT signal to a CMD (stuff too advanced for a weak mind like yours).


> You must be a retard, so sorry everything i tried to explain clearly made no sense, i apologize for that... The only one here not accepting the truth and facts is you, i only tried to point that out to you and all i got back are insults and explicit language, poor you.
You're tough ...
What you explained is to state that in the .cpp there is no trace of the logs folder...
The bugs are such because they lead to unexpected events such as "THE FUCKING OF LOGS FOLDER IS IN SOME WAY COUNTED IN THE SUM".
> 
> I don't have to give you all the answers, hence i am not even the developer of this software and am just like you a end-user

It makes even more laugh that you defined yourself as "end-user" when that role is the user who uses your email and not you who are in mid (since the end user reports his problems to you not on github .. This is if you manage the email service you offer professionally at least ...).


> > Did you go to see what the function does?
> > `SQLCommand command("select sum(messagesize) / 1024 / 1024 as mailboxsize from hm_messages");`
> 
> I already referred to that code didn't I? The fact you didn't bother to look at anything i wrote doesn't make you right....clown!

No, you threw it as if everyone knows programming languages and therefore knows how to find the deception 👏👏👏.
You haven't proved it in the slightest with facts and numbers.

Specifically demonstrates WHO occupied more than 1.5gb.
Not that the logs don't hit anything, but WHO is the real PROBLEM in size.
Since you continue to smoke that I told you that I understand that they are not logs, maybe I have to write it down ... THEN TELL ME WHO IS WITH THE FACTS, NOT THE .CPP OF MY COCK

**Comment by RvdHout** [comment]
Bye, bye....i am done with this bullshit, go and harass someone else,

**Comment by RvdHout** [comment]
End-user of the given software....damn, was that so hard to understand?  Who's the big fool now? 
You must be so simple minded you no longer are worth my time...

**Comment by mstenz** [comment]
unbelievable...the source of the problem (there is one in backup system) is not solved then. If i have time probably I will ad another issue to work on it. hms stores when the backup is started, but when it fails (for whatever reason) the status cannot be reset (only by restarting the service). I think this issue should be closed. I don't want really participate in such type of offended wordings dicussion

**Comment by RvdHout** [comment]
> > unbelievable...the source of the problem (there is one in backup system) is not solved then. If i have time probably I will ad another issue to work on it. hms stores when the backup is started, but when it fails (for whatever reason) the status cannot be reset (only by restarting the service). I think this issue should be closed. I don't want really participate in such type of offended wordings dicussion
> 
> it has already been closed for quite a while (I closed it as soon as I noticed that it was going to end in insults and that he was a lost cause). Yes, open another one because this will then be totally eliminated. If you want also mention the bug that is not checked if the previous backup left remnants for "unexpected" reasons

You are aware you where the one starting the insults and use of explicit language? FYI: I have reported you to github

**Comment by OutbackMatt** [comment]
> and I discover that I had left the debug active and the beauty of 144gb of Logs!

In my server, ~ 15 domains, ~120 accounts, some 2500 messages per day, I log debug every day.
I don't log IMAP, but log everything else.
Also, I write a LOT of custom logs, logs by port, logs by domain, a copy of all message headers for accepted messages and much much more.

I zip my logs each month via script, and I have daily logs going back until December 2011 (Yes, nearly 11 years)

My log folder is 6.4 Gb in size

I'm wondering if your backup had enough space left on your hard drive to write to, or if the hard drive was full


Your supposed screen shot showing 144 GB of log files is entitled "minidump_2022-09-14..." - that's not a log file, it appears to be a zip file perhaps. That's not something that would be created by hMailserver. Did you create this? What makes up those 1230 files?

And what then is the 'Logs.7z'? Is that another 7Z file that you have created? 
hMailserver backup files are named in the form 'HMBackup YYYY-MM-DD ttttt.7z"

And clearly, your current hMailserver.log is only 400 KB

Also, you don't seem to be emptying your 'hmailserver_awstats.log', which would happen automatically if you were using that file in AWStats, so I'm wondering WHY you are logging AWStats?


Looking at the partial logs that you supply, I suspect the real issue for you is the file located at 
"C:\Program Files (x86)\hMailServer\Data\xxxxxxxxxxxxxx\admin\00\{008EFD9E-8357-4493-ADC7-F44389190137}.eml"

I'll bet that is loaded with malware or a virus, and that your system Antivirus software crashes hMailserver when hMailserver tries to copy that file. I'll bet that you haven't excluded the hMailserver data directory from Anti-Virus Scanning...

hMailserver also has some VERY handy built in events that can be used to send an email when a backup has succeeded or indeed failed. I would look into using those moving forwards.

**Comment by RvdHout** [comment]
You must feel very proud of yourself... 

I only tried to proof it wasn't the log files that was the reason the backup failed and all i got in return was a big rant about who knows what using explicit language, I'll rest my case, I am done with you and your little problem or bug as you call it!
